core/src/main/scala/org/apache/spark/scheduler/AsyncEventQueue.scala

```diff
@@ -167,22 +170,19 @@ private class AsyncEventQueue(
     }
     logTrace(s"Dropping event $event")
 
-    val droppedCount = droppedEventsCounter.get
-    if (droppedCount > 0) {
-      // Don't log too frequently
-      if (System.currentTimeMillis() - lastReportTimestamp >= 60 * 1000) {
-        // There may be multiple threads trying to decrease droppedEventsCounter.
-        // Use "compareAndSet" to make sure only one thread can win.
-        // And if another thread is increasing droppedEventsCounter, "compareAndSet" will fail and
-        // then that thread will update it.
-        if (droppedEventsCounter.compareAndSet(droppedCount, 0)) {
-          val prevLastReportTimestamp = lastReportTimestamp
-          lastReportTimestamp = System.currentTimeMillis()
-          val previous = new java.util.Date(prevLastReportTimestamp)
+    val droppedCount = droppedEventsCounter.get - lastDroppedEventsCounter
+    val lastReportTime = lastReportTimestamp.get
+    val curTime = System.currentTimeMillis()
+    // Don't log too frequently
+    if (droppedCount > 0 && curTime - lastReportTime >= LOGGING_INTERVAL) {
+        // There may be multiple threads trying to logging dropped events,
+        // Use 'compareAndSet' to make sure only one thread can win.
+        if (lastReportTimestamp.compareAndSet(lastReportTime, curTime)) {
+          val previous = new java.util.Date(lastReportTime)
+          lastDroppedEventsCounter = droppedCount
```

Comment:
IIUC this should be reset to `droppedEventsCounter.get` ?

Reply:
Nice finding. I'd use a variable to store `droppedEventsCounter.get` before calculating droppedCount and use it for consistency.

Reply:
@jiangxb1987 @HeartSaVioR Thanks, I've fixed it and I think it's more clear to store `droppedEventsCounter.get` before calculating droppedCount.